src/gtktest.rs

```diff
@@ -150,6 +151,24 @@ fn main() {
         format!("<{:.*}>", digits, value)
     });
 
+    spin_button.set_update_policy(SpinButtonUpdatePolicy::IfValid);
+    spin_button.connect_input(|spin_button| {
+        let text = spin_button.get_text().unwrap();
+        println!("spin_button_input: \"{}\"", text);
+        match text.parse::<f64>() {
+            Ok(value) if value >= 90. => {
+                println!("circular right");
+                Some(Ok(10.))
+            }
+            Ok(value) if value <= 10. => {
+                println!("circular left");
+                Some(Ok(90.))
+            }
+            Ok(_) => None,
```

Comment:
I don't think returning `None` is going to behave differently from `Some(Ok(value))` -- the library will parse the text and produce the same value. So just returning it feels more straightforward.